### PR TITLE
utils: Dont use docker.io when checking if local image exists

### LINKED ISSUE #703: The --stack-registry flag breaks fallback to localhost
**Describe the bug**
The [recent addition](https://github.com/appsody/appsody/commit/cadebc7899d44f8554a4986b3b502aca7364da8a) of the `--stack-registry` flag has the side effect of breaking the fallback to localhost. [This line](https://github.com/appsody/appsody/commit/cadebc7899d44f8554a4986b3b502aca7364da8a#diff-b4695c7b6902e27273329af6261b0c73R490) prepends the stack name with the registry host  `docker.io` if the `--stack-registry` flag is unset. 

This is problematic for local stack development because every subsequent call to [`pullImage()`](https://github.com/appsody/appsody/blob/master/cmd/utils.go#L1529) ends up passing an image name such as `docker.io/dev.local/my-new-stack:0.0.1` which then fails when [`checkDockerImageExistsLocally()`](https://github.com/appsody/appsody/blob/master/cmd/utils.go#L1508-L1523) is called.

**NB** In truth, the `--stack-registry` flag was really only enabled for deploy and not init commands. I stumbled on all of this when I uncommented [this line](https://github.com/appsody/appsody/blob/master/cmd/init.go#L84-L85) which seemed to be inadvertently left commented out on the commit. It could be that it wasn't enabled because of exactly this reason. It's not clear from the commit.

See: https://github.com/appsody/appsody/issues/667
See: https://github.com/appsody/appsody/issues/696

**To Reproduce**

0. Uncomment this line https://github.com/appsody/appsody/blob/master/cmd/init.go#L84-L85
1. Ensure you are using a local build of the `appsody` CLI which includes the commit noted.
2. Create a new stack (`appsody stack create my-stack --copy incubator/nodejs-express`)
3. Package the new stack (`appsody stack package`)
4. Create a new project (`mkdir ~/projects/my-project && cd ~/projects/my-project`)
5. Initialize it with the newly created stack (`appsody init -v dev.local/my-stack`)
6. Watch it fail

**Expected behavior**
When I comment out the call to `OverrideStackRegistry()`, things work as expected.

```console
[Debug] Setting up the development environment for projectDir: /home/lanceball/projects/test-appsody-stack and platform: dev.local/ce-functions:0.0.1
[Debug] dev.local/ce-functions:0.0.1 image pulled status: false
[Debug] Pull policy IfNotPresent, checking for local image
[Debug] Docker image ls command output: ecf5f94c588b
[Info] Using local cache for image dev.local/ce-functions:0.0.1
```

**Actual behaviour**
With `OverrideStackRegistry()` left in place, when falling back to localhost, it fails to pull the image.

```console
[Debug] Setting up the development environment for projectDir: /home/lanceball/projects/test-appsody-stack and platform: docker.io/dev.local/ce-functions:0.0.1
[Debug] docker.io/dev.local/ce-functions:0.0.1 image pulled status: false
[Debug] Pull policy IfNotPresent, checking for local image
[Debug] Docker image ls docker.io/dev.local/ce-functions:0.0.1 command output: 
[Info] Pulling docker image docker.io/dev.local/ce-functions:0.0.1
[Info] Running command: docker pull docker.io/dev.local/ce-functions:0.0.1
[Info] Error response from daemon: Get https://dev.local/v2/: dial tcp: lookup dev.local: no such host
[Warning] Docker image pull failed: exit status 1
[Warning] The stack init script failed: Could not find the image either in docker hub or locally: docker.io/dev.local/ce-functions:0.0.1
``` 

**Environment Details (please complete the following information):**
 - OS: Linux

**If applicable please specify:**
 - CLI version: master
 - Stack you are using: I'm trying to create one

### PR DESCRIPTION
**Pull request by groeges**
Fixes #703 

Signed-off-by: Steve Groeger <GROEGES@uk.ibm.com>

### CODE REVIEW

**Comment by kylegc** [comment]
I believe this is a duplicate of #702 and #725, but I'll take a closer look at all three approaches and see which parts make sense to keep.

File: cmd/utils.go [hunk 1]
```diff
@@ -1510,6 +1510,14 @@ func pullCmd(log *LoggingConfig, imageToPull string, buildah bool, dryrun bool)
 
 func checkDockerImageExistsLocally(log *LoggingConfig, imageToPull string) bool {
 	cmdName := "docker"
+
+	imageNameComponents := strings.Split(imageToPull, "/")
+	if len(imageNameComponents) == 3 {
+		if imageNameComponents[0] == "docker.io" {
```
Comment (kylegc) [inline cmd/utils.go]: @groeges Please add an OR "index.docker.io" here

**Comment by Kamran64** [comment]
I've tested this locally and it works for me :)

File: cmd/utils.go (hunk 1 above)
Comment (groeges) [reply cmd/utils.go]: Done

**Pull request merged**